dashboard/queries.py

```diff
@@ -1,14 +1,131 @@
+            query = Session.query.filter(and_(func.upper(Session.name) ==
+                    ident.get_full_subjectid_with_timepoint(),
+                    Session.num == ident.session))
+            if not query.count():
+                ident.session = None
+
+        if not ident.session:
+            query = Session.query.filter((func.upper(Session.name) ==
+                    ident.get_full_subjectid_with_timepoint()))
+
+    return query.all()
+
+
+def get_scan(scan_name, timepoint=None, session=None):
+    """
+    Used by datman. Return a list of matching scans or an empty list
+    """
+    query = Scan.query.filter(Scan.name == scan_name)
+    if timepoint:
+        query = query.filter(Scan.timepoint == timepoint)
+    if session:
+        query = query.filter(Scan.repeat == session)
+    return query.all()
+
+
+def find_scans(search_str):
+    """
+    Used by the dashboard's search bar and so must work around fuzzy user
+    input.
+    """
+    search_str = search_str.strip().upper()
+    try:
+        ident, tag, series, _ = scanid.parse_filename(search_str)
+    except:
+        try:
+            ident = scanid.parse(search_str)
+        except:
+            # Doesnt match a file name or a subject ID so fuzzy search
+            # for...
+            # matching scan name
+            query = Scan.query.filter(func.upper(Scan.name).contains(
+                    search_str))
+            if query.count() == 0:
+                # or matching subid
+                query = Scan.query.filter(func.upper(Scan.timepoint).contains(
+                        search_str))
+            if query.count() == 0:
+                # or matching tags
+                query = Scan.query.filter(func.upper(Scan.tag).contains(
+                        search_str))
+            if query.count() == 0:
+                # or matching series description
+                query = Scan.query.filter(func.upper(Scan.description).contains(
+                        search_str))
+        else:
+            if ident.session:
+                query = Scan.query.filter(and_(func.upper(Scan.timepoint) ==
+                        ident.get_full_subjectid_with_timepoint(),
+                        Scan.repeat == int(ident.session)))
+                if not query.count():
+                    ident.session = None
+
+            if not ident.session:
+                query = Scan.query.filter((func.upper(Scan.timepoint) ==
+                        ident.get_full_subjectid_with_timepoint()))
+    else:
+        name = "_".join([ident.get_full_subjectid_with_timepoint_session(), tag, series])
+        query = Scan.query.filter(func.upper(Scan.name).contains(name))
+
+    return query.all()
+
+def get_user(username):
+    query = User.query.filter(func.lower(User._username).contains(
+            func.lower(username)))
+    return query.all()
 
 def query_metric_values_byid(**kwargs):
     """Queries the database for metrics matching the specifications.
```

Comment:
Oh yeah. All those 'metric_values' functions are legacy stuff that I didn't bother updating because it was already broken and we're planning to overhaul it. The dashboard doesn't crash or anything because of it, those features just dont work (but they didn't really before)